packages/peregrine/lib/util/deriveErrorMessage.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * A function to derive an error string from an array of errors.
+ */
+export const deriveErrorMessage = errors => {
+    const errorCollection = errors.filter(Boolean).map(error => {
+        const { graphQLErrors, message } = error;
+
+        return graphQLErrors && graphQLErrors.length
+            ? graphQLErrors.map(({ message }) => message).join(', ')
+            : message;
```

Comment:
This would be a great place to map from Magento GQL Error string to a more user friendly string.